odxtools/compumethods/texttablecompumethod.py

```diff
@@ -25,17 +28,27 @@ def __post_init__(self) -> None:
     def category(self) -> CompuMethodCategory:
         return "TEXTTABLE"
 
+    def _get_scales(self):
+        scales = list(self.internal_to_phys)
+        if self.compu_default_value:
+            # Default is last, since it's a fallback
+            scales.append(self.compu_default_value)
+        return scales
+
     def convert_physical_to_internal(self, physical_value):
-        scale = next(
-            filter(lambda scale: scale.compu_const == physical_value, self.internal_to_phys), None)
+        scale: CompuScale = next(
+            filter(lambda scale: scale.compu_const == physical_value, self._get_scales()), None)
         if scale is not None:
             res = (
                 scale.compu_inverse_value
                 if scale.compu_inverse_value is not None else scale.lower_limit.value)
             odxassert(self.internal_type.isinstance(res))
             return res
+        raise EncodeError(f"Texttable compu method could not encode '{physical_value}'.")
 
     def __is_internal_in_scale(self, internal_value, scale: CompuScale):
+        if scale == self.compu_default_value:
```

Comment:
is it possible that the default value is not one of the specified scales? (that would be weird, but it would not be the first time...)

Reply:
Yes, it is possible, usually compu_default_value is used to auto assign "invalid", "unknown" to invalid internal values

Reply:
okay, I assumed that in this case "invalid" was simply one of the entries of the table. never stop learning, I suppose ;)